### ISSUE TITLE
[Developer] Del the Phone Layout, But Its Code Block Still There

### ISSUE THREAD

**Issue by MakaraSok** [keymanapp/keyman#1829]
**Describe the bug**
Usually Keyman Dev would generate two touch layouts when one imports into Touch Layout from OSK. When deleting one of those--say Phone--leaving only Tablet, the thing is even then the Phone layout still exists in the Code tab.

**To Reproduce**
Say there are two touch layouts, Tablet and Phone. You wanna get rid of one of the two. 
1. Go to 'Touch Layout' > Design tab; click on the Platform drop down menu then del phone
2. Click on 'Code Tab'; now you should see both codes of both Tablet and Phone

And when you go back to Design Tab, to your surprise the phone layout is there too.

**Expected behavior**
Code block for Phone touch layout should not still be there after being deleted.

**Screenshots**
![2019-06-12_LI](https://user-images.githubusercontent.com/28331388/59332586-c9770080-8d20-11e9-8c17-419a4ba4d926.jpg)
![2019-06-12 (1)_LI](https://user-images.githubusercontent.com/28331388/59333013-ba448280-8d21-11e9-9e7c-fabd6880bdb3.jpg)

---

**Keyman Developer (please complete the following information):
 - OS: Windows 10 1809
 - Keyman Developer version: 11.0.1356.0
 
**Additional context**
The keyboard itself was imported from Windows System keyboard.

**Comment by DavidLRowe** [comment]
Similar problem (though I was trying to delete the "tablet" layout)
```
Keyman Developer 12.0.26.0
Steps to reproduce:
Project, New Project, Basic
Keyboard name: xyz
BCP 47 code: wwa-Latn
OK (taking other defaults)
Click Keyboards tab at bottom, then click xyz.kmn
Click Touch Layout on left
For Platform, "tablet" is displayed. Clicking on down arrow shows "tablet" and "phone"
With "tablet" displayed, click on "Del" button.
"tablet" layout is deleted, display switches to "phone"
Click on "Code" at bottom to see that "tablet" is gone. 
Click on "phone" next to Platform to see that "tablet" is gone.
Ctrl+S to save (* next to xyz.kmn disappears)
Click on "X" next to xyz.kmn to close. xyz.kpj is displayed
Click on xyz.kmn to open
Click on Touch Layout
Notice that Platform set to "tablet"
Clicking on down arrow next to Platform shows both "tablet" and "phone"
```

**Comment by DavidLRowe** [comment]
With above instructions, insert the following before the "Ctrl+S to save":
Make a change to the keyboard (for example, changing the left padding on the space bar to 1)
After saving, undo that change (for example, delete the 1 in the left padding field inserted above)
Perhaps the issue is due to the "Del" button action failing to note that the keyboard has changed and needs to be saved?

**Comment by mcdurdin** [comment]
Fixed by #2114.
